internal/storage/leveldb.go

```diff
@@ -126,13 +166,20 @@ func (s *LevelDBStorage) SetTxStatus(queryID uint64, hash string, neutronHash st
 		return fmt.Errorf("failed to set tx txInfo: %w", err)
 	}
 
+	if processedTx != nil {
```

Comment:
I find this tricky. let's create a `CacheTx` method for the `Storage` interface, huh? this will make `SetTxStatus` more clear and add sense to the `GetCachedTx` method